### PR TITLE
Update Jetty images for latest entrypoint changes

### PR DESCRIPTION
**Pull request by gregw**
improve the docker-entry-point.sh and generate-jetty-start.sh scripts to
be more consistent and to remove the need to switch back to root USER
for configuration generation.

### CODE REVIEW

**Comment by gregw** [comment]
@md5

**Comment by tianon** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/jetty_9.2-jre8/docker-entrypoint.sh b/jetty_9.2-jre8/docker-entrypoint.sh
index 677c323..cd8f821 100755
--- a/jetty_9.2-jre8/docker-entrypoint.sh
+++ b/jetty_9.2-jre8/docker-entrypoint.sh
@@ -26,10 +26,11 @@ if ! command -v -- "$1" >/dev/null 2>&1 ; then
 	set -- java -jar "$JETTY_HOME/start.jar" "$@"
 fi
 
-if [ -z "$TMPDIR" ] ; then
-	TMPDIR=/tmp/jetty
-	mkdir $TMPDIR 2>/dev/null
-fi
+: ${TMPDIR:=/tmp/jetty}
+[ -d "$TMPDIR" ] || mkdir -p $TMPDIR 2>/dev/null
+
+: ${JETTY_START:=$JETTY_BASE/jetty.start}
+
 case "$JAVA_OPTIONS" in
 	*-Djava.io.tmpdir=*) ;;
 	*) JAVA_OPTIONS="-Djava.io.tmpdir=$TMPDIR $JAVA_OPTIONS" ;;
@@ -62,30 +63,40 @@ if expr "$*" : 'java .*/start\.jar.*$' >/dev/null ; then
 		esac
 	done
 
-	if [ -f /jetty-start ] ; then
-		if [ $JETTY_BASE/start.d -nt /jetty-start ] ; then
-			cat >&2 <<- 'EOWARN'
+	if [ $(whoami) != "jetty" ]; then
+		cat >&2 <<- EOWARN
+			********************************************************************
+			WARNING: User is $(whoami)
+			         The user should be (re)set to 'jetty' in the Dockerfile
+			********************************************************************
+		EOWARN
+	fi
+
+	if [ -f $JETTY_START ] ; then
+		if [ $JETTY_BASE/start.d -nt $JETTY_START ] ; then
+			cat >&2 <<- EOWARN
 			********************************************************************
 			WARNING: The $JETTY_BASE/start.d directory has been modified since
-			         the /jetty-start files was generated. Please either delete 
-			         the /jetty-start file or re-run /generate-jetty-start.sh 
+			         the $JETTY_START files was generated. Either delete 
+			         the $JETTY_START file or re-run 
+			             /generate-jetty.start.sh 
 			         from a Dockerfile
 			********************************************************************
 			EOWARN
 		fi
-		echo $(date +'%Y-%m-%d %H:%M:%S.000'):INFO:docker-entrypoint:jetty start command from /jetty-start
-		set -- $(cat /jetty-start)
+		echo $(date +'%Y-%m-%d %H:%M:%S.000'):INFO:docker-entrypoint:jetty start from $JETTY_START
+		set -- $(cat $JETTY_START)
 	else
 		# Do a jetty dry run to set the final command
-		"$@" --dry-run > /$TMPDIR/jetty-start
-		if [ $(egrep -v '\\$' $TMPDIR/jetty-start | wc -l ) -gt 1 ] ; then
+		"$@" --dry-run > $JETTY_START
+		if [ $(egrep -v '\\$' $JETTY_START | wc -l ) -gt 1 ] ; then
 			# command was more than a dry-run
-			cat $TMPDIR/jetty-start \
+			cat $JETTY_START \
 			| awk '/\\$/ { printf "%s", substr($0, 1, length($0)-1); next } 1' \
 			| egrep -v '[^ ]*java .* org\.eclipse\.jetty\.xml\.XmlConfiguration '
 			exit
 		fi
-		set -- $(sed 's/\\$//' $TMPDIR/jetty-start)
+		set -- $(sed 's/\\$//' $JETTY_START)
 	fi
 fi
 
diff --git a/jetty_9.2-jre8/generate-jetty-start.sh b/jetty_9.2-jre8/generate-jetty-start.sh
index 0dbaba7..de78102 100755
--- a/jetty_9.2-jre8/generate-jetty-start.sh
+++ b/jetty_9.2-jre8/generate-jetty-start.sh
@@ -1,3 +1,7 @@
 #!/bin/sh
-rm -f /jetty-start
-/docker-entrypoint.sh --dry-run | sed 's/\\$//' > /jetty-start
+
+if [ -z "$JETTY_START" ] ; then
+	JETTY_START=$JETTY_BASE/jetty.start
+fi
+rm -f $JETTY_START
+/docker-entrypoint.sh --dry-run | sed 's/\\$//' > $JETTY_START
diff --git a/jetty_9.3-jre8-alpine/Dockerfile b/jetty_9.3-jre8-alpine/Dockerfile
index 86d60f1..b1c62d9 100644
--- a/jetty_9.3-jre8-alpine/Dockerfile
+++ b/jetty_9.3-jre8-alpine/Dockerfile
@@ -72,7 +72,7 @@ RUN set -xe \
 	&& mkdir -p "$TMPDIR" \
 	&& chown -R jetty:jetty "$TMPDIR"
 
-COPY docker-entrypoint.sh /
+COPY docker-entrypoint.sh generate-jetty-start.sh /
 
 USER jetty
 EXPOSE 8080
diff --git a/jetty_9.3-jre8-alpine/docker-entrypoint.sh b/jetty_9.3-jre8-alpine/docker-entrypoint.sh
index 677c323..cd8f821 100755
--- a/jetty_9.3-jre8-alpine/docker-entrypoint.sh
+++ b/jetty_9.3-jre8-alpine/docker-entrypoint.sh
@@ -26,10 +26,11 @@ if ! command -v -- "$1" >/dev/null 2>&1 ; then
 	set -- java -jar "$JETTY_HOME/start.jar" "$@"
 fi
 
-if [ -z "$TMPDIR" ] ; then
-	TMPDIR=/tmp/jetty
-	mkdir $TMPDIR 2>/dev/null
-fi
+: ${TMPDIR:=/tmp/jetty}
+[ -d "$TMPDIR" ] || mkdir -p $TMPDIR 2>/dev/null
+
+: ${JETTY_START:=$JETTY_BASE/jetty.start}
+
 case "$JAVA_OPTIONS" in
 	*-Djava.io.tmpdir=*) ;;
 	*) JAVA_OPTIONS="-Djava.io.tmpdir=$TMPDIR $JAVA_OPTIONS" ;;
@@ -62,30 +63,40 @@ if expr "$*" : 'java .*/start\.jar.*$' >/dev/null ; then
 		esac
 	done
 
-	if [ -f /jetty-start ] ; then
-		if [ $JETTY_BASE/start.d -nt /jetty-start ] ; then
-			cat >&2 <<- 'EOWARN'
+	if [ $(whoami) != "jetty" ]; then
+		cat >&2 <<- EOWARN
+			********************************************************************
+			WARNING: User is $(whoami)
+			         The user should be (re)set to 'jetty' in the Dockerfile
+			********************************************************************
+		EOWARN
+	fi
+
+	if [ -f $JETTY_START ] ; then
+		if [ $JETTY_BASE/start.d -nt $JETTY_START ] ; then
+			cat >&2 <<- EOWARN
 			********************************************************************
 			WARNING: The $JETTY_BASE/start.d directory has been modified since
-			         the /jetty-start files was generated. Please either delete 
-			         the /jetty-start file or re-run /generate-jetty-start.sh 
+			         the $JETTY_START files was generated. Either delete 
+			         the $JETTY_START file or re-run 
+			             /generate-jetty.start.sh 
 			         from a Dockerfile
 			********************************************************************
 			EOWARN
 		fi
-		echo $(date +'%Y-%m-%d %H:%M:%S.000'):INFO:docker-entrypoint:jetty start command from /jetty-start
-		set -- $(cat /jetty-start)
+		echo $(date +'%Y-%m-%d %H:%M:%S.000'):INFO:docker-entrypoint:jetty start from $JETTY_START
+		set -- $(cat $JETTY_START)
 	else
 		# Do a jetty dry run to set the final command
-		"$@" --dry-run > /$TMPDIR/jetty-start
-		if [ $(egrep -v '\\$' $TMPDIR/jetty-start | wc -l ) -gt 1 ] ; then
+		"$@" --dry-run > $JETTY_START
+		if [ $(egrep -v '\\$' $JETTY_START | wc -l ) -gt 1 ] ; then
 			# command was more than a dry-run
-			cat $TMPDIR/jetty-start \
+			cat $JETTY_START \
 			| awk '/\\$/ { printf "%s", substr($0, 1, length($0)-1); next } 1' \
 			| egrep -v '[^ ]*java .* org\.eclipse\.jetty\.xml\.XmlConfiguration '
 			exit
 		fi
-		set -- $(sed 's/\\$//' $TMPDIR/jetty-start)
+		set -- $(sed 's/\\$//' $JETTY_START)
 	fi
 fi
 
diff --git a/jetty_9.3-jre8-alpine/generate-jetty-start.sh b/jetty_9.3-jre8-alpine/generate-jetty-start.sh
new file mode 100755
index 0000000..de78102
--- /dev/null
+++ b/jetty_9.3-jre8-alpine/generate-jetty-start.sh
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+if [ -z "$JETTY_START" ] ; then
+	JETTY_START=$JETTY_BASE/jetty.start
+fi
+rm -f $JETTY_START
+/docker-entrypoint.sh --dry-run | sed 's/\\$//' > $JETTY_START
diff --git a/jetty_9.3-jre8/docker-entrypoint.sh b/jetty_9.3-jre8/docker-entrypoint.sh
index 677c323..cd8f821 100755
--- a/jetty_9.3-jre8/docker-entrypoint.sh
+++ b/jetty_9.3-jre8/docker-entrypoint.sh
@@ -26,10 +26,11 @@ if ! command -v -- "$1" >/dev/null 2>&1 ; then
 	set -- java -jar "$JETTY_HOME/start.jar" "$@"
 fi
 
-if [ -z "$TMPDIR" ] ; then
-	TMPDIR=/tmp/jetty
-	mkdir $TMPDIR 2>/dev/null
-fi
+: ${TMPDIR:=/tmp/jetty}
+[ -d "$TMPDIR" ] || mkdir -p $TMPDIR 2>/dev/null
+
+: ${JETTY_START:=$JETTY_BASE/jetty.start}
+
 case "$JAVA_OPTIONS" in
 	*-Djava.io.tmpdir=*) ;;
 	*) JAVA_OPTIONS="-Djava.io.tmpdir=$TMPDIR $JAVA_OPTIONS" ;;
@@ -62,30 +63,40 @@ if expr "$*" : 'java .*/start\.jar.*$' >/dev/null ; then
 		esac
 	done
 
-	if [ -f /jetty-start ] ; then
-		if [ $JETTY_BASE/start.d -nt /jetty-start ] ; then
-			cat >&2 <<- 'EOWARN'
+	if [ $(whoami) != "jetty" ]; then
+		cat >&2 <<- EOWARN
+			********************************************************************
+			WARNING: User is $(whoami)
+			         The user should be (re)set to 'jetty' in the Dockerfile
+			********************************************************************
+		EOWARN
+	fi
+
+	if [ -f $JETTY_START ] ; then
+		if [ $JETTY_BASE/start.d -nt $JETTY_START ] ; then
+			cat >&2 <<- EOWARN
 			********************************************************************
 			WARNING: The $JETTY_BASE/start.d directory has been modified since
-			         the /jetty-start files was generated. Please either delete 
-			         the /jetty-start file or re-run /generate-jetty-start.sh 
+			         the $JETTY_START files was generated. Either delete 
+			         the $JETTY_START file or re-run 
+			             /generate-jetty.start.sh 
 			         from a Dockerfile
 			********************************************************************
 			EOWARN
 		fi
-		echo $(date +'%Y-%m-%d %H:%M:%S.000'):INFO:docker-entrypoint:jetty start command from /jetty-start
-		set -- $(cat /jetty-start)
+		echo $(date +'%Y-%m-%d %H:%M:%S.000'):INFO:docker-entrypoint:jetty start from $JETTY_START
+		set -- $(cat $JETTY_START)
 	else
 		# Do a jetty dry run to set the final command
-		"$@" --dry-run > /$TMPDIR/jetty-start
-		if [ $(egrep -v '\\$' $TMPDIR/jetty-start | wc -l ) -gt 1 ] ; then
+		"$@" --dry-run > $JETTY_START
+		if [ $(egrep -v '\\$' $JETTY_START | wc -l ) -gt 1 ] ; then
 			# command was more than a dry-run
-			cat $TMPDIR/jetty-start \
+			cat $JETTY_START \
 			| awk '/\\$/ { printf "%s", substr($0, 1, length($0)-1); next } 1' \
 			| egrep -v '[^ ]*java .* org\.eclipse\.jetty\.xml\.XmlConfiguration '
 			exit
 		fi
-		set -- $(sed 's/\\$//' $TMPDIR/jetty-start)
+		set -- $(sed 's/\\$//' $JETTY_START)
 	fi
 fi
 
diff --git a/jetty_9.3-jre8/generate-jetty-start.sh b/jetty_9.3-jre8/generate-jetty-start.sh
index 0dbaba7..de78102 100755
--- a/jetty_9.3-jre8/generate-jetty-start.sh
+++ b/jetty_9.3-jre8/generate-jetty-start.sh
@@ -1,3 +1,7 @@
 #!/bin/sh
-rm -f /jetty-start
-/docker-entrypoint.sh --dry-run | sed 's/\\$//' > /jetty-start
+
+if [ -z "$JETTY_START" ] ; then
+	JETTY_START=$JETTY_BASE/jetty.start
+fi
+rm -f $JETTY_START
+/docker-entrypoint.sh --dry-run | sed 's/\\$//' > $JETTY_START
diff --git a/jetty_jre7/docker-entrypoint.sh b/jetty_jre7/docker-entrypoint.sh
index 677c323..cd8f821 100755
--- a/jetty_jre7/docker-entrypoint.sh
+++ b/jetty_jre7/docker-entrypoint.sh
@@ -26,10 +26,11 @@ if ! command -v -- "$1" >/dev/null 2>&1 ; then
 	set -- java -jar "$JETTY_HOME/start.jar" "$@"
 fi
 
-if [ -z "$TMPDIR" ] ; then
-	TMPDIR=/tmp/jetty
-	mkdir $TMPDIR 2>/dev/null
-fi
+: ${TMPDIR:=/tmp/jetty}
+[ -d "$TMPDIR" ] || mkdir -p $TMPDIR 2>/dev/null
+
+: ${JETTY_START:=$JETTY_BASE/jetty.start}
+
 case "$JAVA_OPTIONS" in
 	*-Djava.io.tmpdir=*) ;;
 	*) JAVA_OPTIONS="-Djava.io.tmpdir=$TMPDIR $JAVA_OPTIONS" ;;
@@ -62,30 +63,40 @@ if expr "$*" : 'java .*/start\.jar.*$' >/dev/null ; then
 		esac
 	done
 
-	if [ -f /jetty-start ] ; then
-		if [ $JETTY_BASE/start.d -nt /jetty-start ] ; then
-			cat >&2 <<- 'EOWARN'
+	if [ $(whoami) != "jetty" ]; then
+		cat >&2 <<- EOWARN
+			********************************************************************
+			WARNING: User is $(whoami)
+			         The user should be (re)set to 'jetty' in the Dockerfile
+			********************************************************************
+		EOWARN
+	fi
+
+	if [ -f $JETTY_START ] ; then
+		if [ $JETTY_BASE/start.d -nt $JETTY_START ] ; then
+			cat >&2 <<- EOWARN
 			********************************************************************
 			WARNING: The $JETTY_BASE/start.d directory has been modified since
-			         the /jetty-start files was generated. Please either delete 
-			         the /jetty-start file or re-run /generate-jetty-start.sh 
+			         the $JETTY_START files was generated. Either delete 
+			         the $JETTY_START file or re-run 
+			             /generate-jetty.start.sh 
 			         from a Dockerfile
 			********************************************************************
 			EOWARN
 		fi
-		echo $(date +'%Y-%m-%d %H:%M:%S.000'):INFO:docker-entrypoint:jetty start command from /jetty-start
-		set -- $(cat /jetty-start)
+		echo $(date +'%Y-%m-%d %H:%M:%S.000'):INFO:docker-entrypoint:jetty start from $JETTY_START
+		set -- $(cat $JETTY_START)
 	else
 		# Do a jetty dry run to set the final command
-		"$@" --dry-run > /$TMPDIR/jetty-start
-		if [ $(egrep -v '\\$' $TMPDIR/jetty-start | wc -l ) -gt 1 ] ; then
+		"$@" --dry-run > $JETTY_START
+		if [ $(egrep -v '\\$' $JETTY_START | wc -l ) -gt 1 ] ; then
 			# command was more than a dry-run
-			cat $TMPDIR/jetty-start \
+			cat $JETTY_START \
 			| awk '/\\$/ { printf "%s", substr($0, 1, length($0)-1); next } 1' \
 			| egrep -v '[^ ]*java .* org\.eclipse\.jetty\.xml\.XmlConfiguration '
 			exit
 		fi
-		set -- $(sed 's/\\$//' $TMPDIR/jetty-start)
+		set -- $(sed 's/\\$//' $JETTY_START)
 	fi
 fi
 
diff --git a/jetty_jre7/generate-jetty-start.sh b/jetty_jre7/generate-jetty-start.sh
index 0dbaba7..de78102 100755
--- a/jetty_jre7/generate-jetty-start.sh
+++ b/jetty_jre7/generate-jetty-start.sh
@@ -1,3 +1,7 @@
 #!/bin/sh
-rm -f /jetty-start
-/docker-entrypoint.sh --dry-run | sed 's/\\$//' > /jetty-start
+
+if [ -z "$JETTY_START" ] ; then
+	JETTY_START=$JETTY_BASE/jetty.start
+fi
+rm -f $JETTY_START
+/docker-entrypoint.sh --dry-run | sed 's/\\$//' > $JETTY_START
diff --git a/jetty_jre8-alpine/Dockerfile b/jetty_jre8-alpine/Dockerfile
index 2548e30..54856d9 100644
--- a/jetty_jre8-alpine/Dockerfile
+++ b/jetty_jre8-alpine/Dockerfile
@@ -66,7 +66,7 @@ RUN set -xe \
 	&& mkdir -p "$TMPDIR" \
 	&& chown -R jetty:jetty "$TMPDIR"
 
-COPY docker-entrypoint.sh /
+COPY docker-entrypoint.sh generate-jetty-start.sh /
 
 USER jetty
 EXPOSE 8080
diff --git a/jetty_jre8-alpine/docker-entrypoint.sh b/jetty_jre8-alpine/docker-entrypoint.sh
index 677c323..cd8f821 100755
--- a/jetty_jre8-alpine/docker-entrypoint.sh
+++ b/jetty_jre8-alpine/docker-entrypoint.sh
@@ -26,10 +26,11 @@ if ! command -v -- "$1" >/dev/null 2>&1 ; then
 	set -- java -jar "$JETTY_HOME/start.jar" "$@"
 fi
 
-if [ -z "$TMPDIR" ] ; then
-	TMPDIR=/tmp/jetty
-	mkdir $TMPDIR 2>/dev/null
-fi
+: ${TMPDIR:=/tmp/jetty}
+[ -d "$TMPDIR" ] || mkdir -p $TMPDIR 2>/dev/null
+
+: ${JETTY_START:=$JETTY_BASE/jetty.start}
+
 case "$JAVA_OPTIONS" in
 	*-Djava.io.tmpdir=*) ;;
 	*) JAVA_OPTIONS="-Djava.io.tmpdir=$TMPDIR $JAVA_OPTIONS" ;;
@@ -62,30 +63,40 @@ if expr "$*" : 'java .*/start\.jar.*$' >/dev/null ; then
 		esac
 	done
 
-	if [ -f /jetty-start ] ; then
-		if [ $JETTY_BASE/start.d -nt /jetty-start ] ; then
-			cat >&2 <<- 'EOWARN'
+	if [ $(whoami) != "jetty" ]; then
+		cat >&2 <<- EOWARN
+			********************************************************************
+			WARNING: User is $(whoami)
+			         The user should be (re)set to 'jetty' in the Dockerfile
+			********************************************************************
+		EOWARN
+	fi
+
+	if [ -f $JETTY_START ] ; then
+		if [ $JETTY_BASE/start.d -nt $JETTY_START ] ; then
+			cat >&2 <<- EOWARN
 			********************************************************************
 			WARNING: The $JETTY_BASE/start.d directory has been modified since
-			         the /jetty-start files was generated. Please either delete 
-			         the /jetty-start file or re-run /generate-jetty-start.sh 
+			         the $JETTY_START files was generated. Either delete 
+			         the $JETTY_START file or re-run 
+			             /generate-jetty.start.sh 
 			         from a Dockerfile
 			********************************************************************
 			EOWARN
 		fi
-		echo $(date +'%Y-%m-%d %H:%M:%S.000'):INFO:docker-entrypoint:jetty start command from /jetty-start
-		set -- $(cat /jetty-start)
+		echo $(date +'%Y-%m-%d %H:%M:%S.000'):INFO:docker-entrypoint:jetty start from $JETTY_START
+		set -- $(cat $JETTY_START)
 	else
 		# Do a jetty dry run to set the final command
-		"$@" --dry-run > /$TMPDIR/jetty-start
-		if [ $(egrep -v '\\$' $TMPDIR/jetty-start | wc -l ) -gt 1 ] ; then
+		"$@" --dry-run > $JETTY_START
+		if [ $(egrep -v '\\$' $JETTY_START | wc -l ) -gt 1 ] ; then
 			# command was more than a dry-run
-			cat $TMPDIR/jetty-start \
+			cat $JETTY_START \
 			| awk '/\\$/ { printf "%s", substr($0, 1, length($0)-1); next } 1' \
 			| egrep -v '[^ ]*java .* org\.eclipse\.jetty\.xml\.XmlConfiguration '
 			exit
 		fi
-		set -- $(sed 's/\\$//' $TMPDIR/jetty-start)
+		set -- $(sed 's/\\$//' $JETTY_START)
 	fi
 fi
 
diff --git a/jetty_jre8-alpine/generate-jetty-start.sh b/jetty_jre8-alpine/generate-jetty-start.sh
new file mode 100755
index 0000000..de78102
--- /dev/null
+++ b/jetty_jre8-alpine/generate-jetty-start.sh
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+if [ -z "$JETTY_START" ] ; then
+	JETTY_START=$JETTY_BASE/jetty.start
+fi
+rm -f $JETTY_START
+/docker-entrypoint.sh --dry-run | sed 's/\\$//' > $JETTY_START
diff --git a/jetty_jre8/docker-entrypoint.sh b/jetty_jre8/docker-entrypoint.sh
index 677c323..cd8f821 100755
--- a/jetty_jre8/docker-entrypoint.sh
+++ b/jetty_jre8/docker-entrypoint.sh
@@ -26,10 +26,11 @@ if ! command -v -- "$1" >/dev/null 2>&1 ; then
 	set -- java -jar "$JETTY_HOME/start.jar" "$@"
 fi
 
-if [ -z "$TMPDIR" ] ; then
-	TMPDIR=/tmp/jetty
-	mkdir $TMPDIR 2>/dev/null
-fi
+: ${TMPDIR:=/tmp/jetty}
+[ -d "$TMPDIR" ] || mkdir -p $TMPDIR 2>/dev/null
+
+: ${JETTY_START:=$JETTY_BASE/jetty.start}
+
 case "$JAVA_OPTIONS" in
 	*-Djava.io.tmpdir=*) ;;
 	*) JAVA_OPTIONS="-Djava.io.tmpdir=$TMPDIR $JAVA_OPTIONS" ;;
@@ -62,30 +63,40 @@ if expr "$*" : 'java .*/start\.jar.*$' >/dev/null ; then
 		esac
 	done
 
-	if [ -f /jetty-start ] ; then
-		if [ $JETTY_BASE/start.d -nt /jetty-start ] ; then
-			cat >&2 <<- 'EOWARN'
+	if [ $(whoami) != "jetty" ]; then
+		cat >&2 <<- EOWARN
+			********************************************************************
+			WARNING: User is $(whoami)
+			         The user should be (re)set to 'jetty' in the Dockerfile
+			********************************************************************
+		EOWARN
+	fi
+
+	if [ -f $JETTY_START ] ; then
+		if [ $JETTY_BASE/start.d -nt $JETTY_START ] ; then
+			cat >&2 <<- EOWARN
 			********************************************************************
 			WARNING: The $JETTY_BASE/start.d directory has been modified since
-			         the /jetty-start files was generated. Please either delete 
-			         the /jetty-start file or re-run /generate-jetty-start.sh 
+			         the $JETTY_START files was generated. Either delete 
+			         the $JETTY_START file or re-run 
+			             /generate-jetty.start.sh 
 			         from a Dockerfile
 			********************************************************************
 			EOWARN
 		fi
-		echo $(date +'%Y-%m-%d %H:%M:%S.000'):INFO:docker-entrypoint:jetty start command from /jetty-start
-		set -- $(cat /jetty-start)
+		echo $(date +'%Y-%m-%d %H:%M:%S.000'):INFO:docker-entrypoint:jetty start from $JETTY_START
+		set -- $(cat $JETTY_START)
 	else
 		# Do a jetty dry run to set the final command
-		"$@" --dry-run > /$TMPDIR/jetty-start
-		if [ $(egrep -v '\\$' $TMPDIR/jetty-start | wc -l ) -gt 1 ] ; then
+		"$@" --dry-run > $JETTY_START
+		if [ $(egrep -v '\\$' $JETTY_START | wc -l ) -gt 1 ] ; then
 			# command was more than a dry-run
-			cat $TMPDIR/jetty-start \
+			cat $JETTY_START \
 			| awk '/\\$/ { printf "%s", substr($0, 1, length($0)-1); next } 1' \
 			| egrep -v '[^ ]*java .* org\.eclipse\.jetty\.xml\.XmlConfiguration '
 			exit
 		fi
-		set -- $(sed 's/\\$//' $TMPDIR/jetty-start)
+		set -- $(sed 's/\\$//' $JETTY_START)
 	fi
 fi
 
diff --git a/jetty_jre8/generate-jetty-start.sh b/jetty_jre8/generate-jetty-start.sh
index 0dbaba7..de78102 100755
--- a/jetty_jre8/generate-jetty-start.sh
+++ b/jetty_jre8/generate-jetty-start.sh
@@ -1,3 +1,7 @@
 #!/bin/sh
-rm -f /jetty-start
-/docker-entrypoint.sh --dry-run | sed 's/\\$//' > /jetty-start
+
+if [ -z "$JETTY_START" ] ; then
+	JETTY_START=$JETTY_BASE/jetty.start
+fi
+rm -f $JETTY_START
+/docker-entrypoint.sh --dry-run | sed 's/\\$//' > $JETTY_START
````

</details>

File: library/jetty [hunk 1]
```diff
@@ -4,24 +4,25 @@ GitRepo: https://github.com/appropriate/docker-jetty.git
...
 Tags: 9.2.22, 9.2, 9.2.22-jre8, 9.2-jre8
 Directory: 9.2-jre8
-GitCommit: bb1a968360166dec223b0eff1a979e1acad8b976
+GitCommit: 6938dbcf3c3f1e433fca63909400670a52f39dfe
 
 Tags: 9.2.22-jre7, 9.2-jre7, 9-jre7, jre7
 Directory: 9.2-jre7
-GitCommit: bb1a968360166dec223b0eff1a979e1acad8b976
+GitCommit: 6938dbcf3c3f1e433fca63909400670a52f39dfe
+
```
Comment (md5) [inline library/jetty]: It seems weird that we got an extra line break at the end here. I don't see how `generate-stackbrew-library.sh` would do that.

**Comment by tianon** [comment]
LGTM (the extra newline is odd, but harmless and easily removed in a follow-up :+1:)

Build test of #3577; 5f229a52 (`jetty`):

```console

$ bashbrew build jetty:9.4.7
Using bashbrew/cache:c85232ea6524c33c82f4e5f90971024ac4b59c3b6d3c213484316237cdceb34d (jetty:9.4.7)
Tagging jetty:9.4.7
Tagging jetty:9.4
Tagging jetty:9
Tagging jetty:9.4.7-jre8
Tagging jetty:9.4-jre8
Tagging jetty:9-jre8
Tagging jetty:latest
Tagging jetty:jre8

$ test/run.sh jetty:9.4.7
testing jetty:9.4.7
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'jetty-hello-web' [5/5]....passed


$ bashbrew build jetty:9.4.7-alpine
Using bashbrew/cache:a16a8ff700312d36048b81ec7185dc35ffa4113d7549359a429ef29b5f4d3714 (jetty:9.4.7-alpine)
Tagging jetty:9.4.7-alpine
Tagging jetty:9.4-alpine
Tagging jetty:9-alpine
Tagging jetty:9.4.7-jre8-alpine
Tagging jetty:9.4-jre8-alpine
Tagging jetty:9-jre8-alpine
Tagging jetty:alpine
Tagging jetty:jre8-alpine

$ test/run.sh jetty:9.4.7-alpine
testing jetty:9.4.7-alpine
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'jetty-hello-web' [5/5]....passed


$ bashbrew build jetty:9.3.21
Using bashbrew/cache:773a751cf33e4d0dd49c7c3cc96bdc62ba00dc3d0ea7e13f22d23f690ee69694 (jetty:9.3.21)
Tagging jetty:9.3.21
Tagging jetty:9.3
Tagging jetty:9.3.21-jre8
Tagging jetty:9.3-jre8

$ test/run.sh jetty:9.3.21
testing jetty:9.3.21
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'jetty-hello-web' [5/5]....passed


$ bashbrew build jetty:9.3.21-alpine
Using bashbrew/cache:13d23c2a78497653eadc670c31821ebf4d10555cd875d28215ab6fe9e4c73908 (jetty:9.3.21-alpine)
Tagging jetty:9.3.21-alpine
Tagging jetty:9.3-alpine
Tagging jetty:9.3.21-jre8-alpine
Tagging jetty:9.3-jre8-alpine

$ test/run.sh jetty:9.3.21-alpine
testing jetty:9.3.21-alpine
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'jetty-hello-web' [5/5]....passed


$ bashbrew build jetty:9.2.22
Using bashbrew/cache:edc6bffb6de0d2b4d4b348239ccb7055f1f9a209182a792418b8135d00966dd3 (jetty:9.2.22)
Tagging jetty:9.2.22
Tagging jetty:9.2
Tagging jetty:9.2.22-jre8
Tagging jetty:9.2-jre8

$ test/run.sh jetty:9.2.22
testing jetty:9.2.22
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'jetty-hello-web' [5/5].....passed


$ bashbrew build jetty:9.2.22-jre7
Using bashbrew/cache:7bcf54be06103f16dd1bb18bafed61f988b979eb5b2ba1859463f78b26166d56 (jetty:9.2.22-jre7)
Tagging jetty:9.2.22-jre7
Tagging jetty:9.2-jre7
Tagging jetty:9-jre7
Tagging jetty:jre7

$ test/run.sh jetty:9.2.22-jre7
testing jetty:9.2.22-jre7
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'jetty-hello-web' [5/5]......passed

```

**Pull request merged**